src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -675,7 +675,7 @@ private bool TakeStartLine(SocketInput input)
                     pathEnd = UrlPathDecoder.Unescape(pathBegin, pathEnd);
                 }
 
-                var requestUrlPath = pathBegin.GetString(pathEnd);
+                var requestUrlPath = pathBegin.GetUtf8String(pathEnd);
```

Comment:
https://tools.ietf.org/html/rfc3986#page-11 seems to suggest so; but looking for a more recent one...